pkgs/development/tools/ocaml/melange/default.nix

```diff
@@ -0,0 +1,65 @@
+{
```

Comment:
not sure if this file could go into pkgs/by-name. generally we try to move everything there, but there could be exceptions for language specific packages that I'm not aware of

Reply:
I tried to make it as similar as possible to the js_of_ocaml package, which serves a very similar purpose.

Reply:
Note that there is also a kind-of naming conflict with the existing "pkgs.melange" package (completely unrelated package), which is not an actual conflict as this one goes to "pkgs.ocamlPackages.melange".

Reply:
Thinking more of this, should I rename this package to "melange-re" and move it `pkgs/by-name/me/melange-re`?

If so, would it be okay to still be named `ocamlPackages.melange` in the Nix code, or should it be renamed to `ocamlPackages.melange-re` as well?

Reply:
I think the later sounds like a good idea, but I'm not an expert in the ocaml ecosystem :)

Reply:
Ok, will do.

Reply:
Well, melange also works with OCaml 4.14, what should we do about that?

Reply:
I think you've already added it to the right place, but yeah let's add the 4.14 version too